### DISCUSSION TITLE
Add "junit_suite_name" to behave command line arguments

### DISCUSSION THREAD

**Discussion by RehabAbdAllah** [behave/behave#1071] [Ideas]
Hi , 
I have a case where I run the BDD feature files for multiple devices in the test setup ,when viewing the results on jenkins junit reports , all test suites and test cases has the same exact Name , which make it hard to distinguish which results belong to which device .
With junit_suite_name command line , it will be easier to update test suite names with every run .

**Comment by jenisys** [comment]
I am not quite sure if that is a `behave` problem.
Each `testsuite` and `testcase` XML element in the Unit test report provide a `name` and `classname` attribute.

For example:
* `Feature name/title` leads to `<testsuite name="${feature.name}" ...>`
* `Scenario name/title` leads to `<testcase name="${scenario.name}" ...>` 

Therefore, you should clarify if this is a Jenkins issue, a user issue (because you provided no Feature names, etc.) or a `behave` issue.

OTHERWISE:
* Currently, we will not add additional command-line options for `junit_suite_name`.
  I would not know anyway where to put this information, because the JUnit reports are provided per feature file.
  And you should have several feature files.
* We may provide conventions for a Userdata scope that its properties are added as properties in the JUnit XML report.

HINT:
A `junitparser` Python package is provided on https://pypi.org (or look for similar packages/projects).
Maybe a JUnitXML post-processor is the better solution to feed Jenkins a customised or combined report.

**Comment by jenisys** [comment]
`junitparser` allows to merge `behave` `TESTS-*.xml` reports into one JUnit test report file.
In addition, you can provide a `testsuites.@name` attribute value, like:

```shell
$ junitparser merge --suite-name=XYZ reports/TESTS-*.xml behave_combined_report.xml
```
